### PR TITLE
Add `mainImage` config option for change main image

### PR DESCRIPTION
**Pull request by vxcamiloxv**
When anyone run a instance the first action is change the 'some friends' image and exists two PR for change the image, so, let's make everyone happy.

### Related PRs
https://github.com/pump-io/pump.io/pull/867
https://github.com/pump-io/pump.io/pull/1290

### Screenshot
example

### CODE REVIEW

**Comment by vxcamiloxv** [comment]
Yes, better

File: README.md [hunk 1]
```diff
@@ -139,6 +139,8 @@ Here are the main configuration keys.
   "pump.io".
 * *owner* Name of owning entity, if you want to link to it.
 * *ownerURL* URL of owning entity, if you want to link to it.
+* *mainImage* A image/icon for main page, could be a external URL or
```
Comment (strugee) [inline README.md]: Since #1496 will land soon this entire section will be gone soon - can you send an MR to https://gitlab.com/pump.io/pump-docs with this change? And then get rid of these two lines in this PR?

File: public/stylesheet/pumpio.css [hunk 2]
```diff
@@ -12,6 +12,13 @@ body {
   color: white;
 }
 
+/* main */
+
+.hero-unit .span4 img {
+  max-height: 235px;
+  max-width: 340px;
```
Comment (strugee) [inline public/stylesheet/pumpio.css]: My CSS is a little shaky; `max-height` and `max-width` don't cause the image to be stretched or anything like `width`/`height` do, right? Just scaled?

File: public/template/main-client.jade [hunk 3]
```diff
@@ -22,8 +22,9 @@
           p
             a.btn.primary.large(href="/main/register") Get Started &raquo;
 
-      .span4
-        img.img-rounded(src="images/somefriends.jpg", width="340", height="235")
+      .span4.pull-center
+        if config.mainImage
+          img.img-rounded(src="#{config.mainImage}")
```
Comment (strugee) [inline public/template/main-client.jade]: Maybe there's nothing to be done, but AFAICT there'll be a layout reflow now when the image is loaded right? Since the `width` and `height` HTML attributes are now gone?

File: public/stylesheet/pumpio.css [hunk 4]
```diff
@@ -253,6 +260,12 @@ body {
   margin-top: 20px;
 }
 
+/* helpers */
+
+.pull-center {
+  text-align: center;
```
Comment (strugee) [inline public/stylesheet/pumpio.css]: Why is this needed?

File: public/stylesheet/pumpio.css (hunk 2 above)
Comment (vxcamiloxv) [reply public/stylesheet/pumpio.css]: no, the max-(height/width) keep proportions and keep the size when is smaller than the maximum size (not force the size)

File: public/stylesheet/pumpio.css (hunk 4 above)
Comment (vxcamiloxv) [reply public/stylesheet/pumpio.css]: for https://github.com/pump-io/pump.io/pull/1486/files#diff-a81d9634aa223a78b1c4e1e3ddad90ecR25, when the width is less than the maximum it looks to right and it looks very weird, that helper fix and looks like the image.

File: public/template/main-client.jade (hunk 3 above)
Comment (vxcamiloxv) [reply public/template/main-client.jade]: Yes, the width/height in the element is a little hardcoded and force the size, the max-(width/height) fix that.

Comment (strugee) [reply public/template/main-client.jade]: You misunderstood me. The layout of the home page will shift when the image is loaded because the width and height of the image are now only known after the image request completes, not when the HTML loads. Unless I'm missing something?

AFAIK there isn't really anything to be done about this, but it _does_ seem unfortunate...

Comment (vxcamiloxv) [reply public/template/main-client.jade]: No, the behavior is the same, not affect the render times, the css is loaded before the image

**Comment by strugee** [comment]
@vxcamiloxv ping, https://github.com/pump-io/pump.io/pull/1486#discussion_r154546602 is the only thing blocking this from being merged

**Comment by vxcamiloxv** [comment]
@strugee  Done.

File: public/template/main-client.jade (hunk 3 above)
Comment (strugee) [inline public/template/main-client.jade]: Hey, sorry, was all ready to merge this and then I noticed this suuuper small change - this attribute can be simplified to `src=config.mainImage` :thumbsup:

**Comment by strugee** [comment]
Dropping this here so it doesn't fall off my radar: https://gitlab.com/pump.io/pump-docs/merge_requests/9

I also left a comment there.

File: public/template/main-client.jade (hunk 3 above)
Comment (vxcamiloxv) [reply public/template/main-client.jade]: done

**Comment by vxcamiloxv** [comment]
@strugee the test is breaking in 'Error: Cannot find module 'nomnom'' but is not introduced in this PR

**Comment by vxcamiloxv** [comment]
@strugee  works again :)

**Comment by Serkan-devel** [comment]
I think the image source should be changed on the README too

**Comment by vxcamiloxv** [comment]
@Serkan-devel  Is already have done in: https://gitlab.com/pump.io/pump-docs, the configuration was moved from README to pump-docs

**Comment by Serkan-devel** [comment]
But on https://github.com/pump-io/pump.io/blob/master/README.md#L70 the older images is still linked

**Comment by vxcamiloxv** [comment]
@Serkan-devel the default image is the same, This PR only introduce the possibility to change it.

**Comment by Serkan-devel** [comment]
Oh

**Comment by Serkan-devel** [comment]
Then maybe add the source of the other image

**Comment by vxcamiloxv** [comment]
@Serkan-devel does not exist "other" image, is the same image of previous pump.io versions, but now any instance can change it in `pump.io.json`
